### PR TITLE
Add trajectory curve plotting to dittoDimPlot.

### PR DESCRIPTION
**Pull request by j-andrews7**
Ka-pow. Tests pass. Mildly certain I didn't break anything. Renaming of certain arguments for clarity was necessary.

If this is mentioned in any of the docs, let me know and I can go edit them, but a brief ctrl-f on them didn't bring up anything specific for the trajectory stuff.

### CODE REVIEW

**Comment by dtm2451** [comment]
Looks great!  A few suggestions:

1. The documentation about the structure of `add.trajectory.curves` should be updated to reflect that elements of the list are dataframes containing x and y coordinates in columns 1 and 2 respectively.
2. I think the slingshot note should also have dimension control inputs added because the user will need to manually make sure these match the current plot.

File: R/DittoDimPlot.R [hunk 1]
```diff
@@ -74,8 +74,10 @@
...
-#' @param add.trajectories List vectors representing trajectory paths from start-cluster to end-cluster where vector contents are the names of clusters provided in the \code{trajectories.cluster.meta} input.
+#' @param add.trajectory.lineages List of vectors representing trajectory paths from start-cluster to end-cluster where vector contents are the names of clusters provided in the \code{trajectories.cluster.meta} input.
 #' If Slingshot package was used for trajectory analysis, you can use \code{add.trajectories = SlingshotDataSet(SCE_with_slingshot)$lineages}. In future versions, I might build such retrieval in by default for SCEs.
+#' @param add.trajectory.curves List of vectors representing trajectory paths from start to end where vector contents are x, y coordinates.
+#' If Slingshot package was used for trajectory analysis, you can use \code{add.trajectories = SlingshotDataSet(SCE_with_slingshot)$curves}.
```
Comment (dtm2451) [inline R/DittoDimPlot.R]: End with: SlingshotDataSet(SCE_with_slingshot)$curves[,c(dim.1, dim.2)].Note: These curves are not calculated per-dimensionality reduction by dittoSeq, so the dimensions provided to this input need to match dimensions used by dittoDimPlot.

Comment (dtm2451) [reply R/DittoDimPlot.R]: or something like that

Comment (j-andrews7) [reply R/DittoDimPlot.R]: Couldn't do it quite like this, as each element of curves is actually a named list, which contains an element `s` that actually contains the coordinates. Think it should be explicit enough with the changes though.

**Pull request merged**